src/main/java/sage/parser/Parser.java

```diff
@@ -0,0 +1,180 @@
+     * @return A DeleteCommand object.
+     * @throws SageException If the arguments are invalid or missing.
+     */
+    private static Command parseDeleteCommand(String arguments) throws SageException {
+        if (arguments.isEmpty()) {
+            throw new SageException("Please specify the task number to delete.");
+        }
+        int taskIndex = Integer.parseInt(arguments) - 1;
+        return new DeleteCommand(taskIndex);
+    }
+
+    /**
+     * Parses arguments for a 'todo' command and returns an AddTodoCommand object.
+     * @param arguments The arguments string for the 'todo' command.
+     * @return An AddTodoCommand object.
+     * @throws SageException If the arguments are invalid or missing.
+     */
+    private static Command parseTodoCommand(String arguments) throws SageException {
+        if (arguments.trim().isEmpty()) {
+            throw new SageException("The description of a todo cannot be empty.");
+        }
+        return new AddTodoCommand(arguments);
+    }
+
+    /**
+     * Parses arguments for a 'deadline' command and returns an AddDeadlineCommand object.
+     * @param arguments The arguments string for the 'deadline' command.
+     * @return An AddDeadlineCommand object.
+     * @throws SageException If the arguments are invalid or missing, or date format is incorrect.
+     */
+    private static Command parseDeadlineCommand(String arguments) throws SageException {
+        if (arguments.trim().isEmpty()) {
+            throw new SageException("The description of a deadline cannot be empty.");
+        }
+        if (!arguments.contains(" /by ")) {
+            throw new SageException("Invalid deadline format. Please use: deadline <description> /by <d/M/yyyy HHmm>");
+        }
+        String[] deadlineParts = arguments.split(" /by ");
+        LocalDateTime by = parseDateTime(deadlineParts[1]);
+        return new AddDeadlineCommand(deadlineParts[0], by);
+    }
+
+    /**
+     * Parses arguments for an 'event' command and returns an AddEventCommand object.
+     * @param arguments The arguments string for the 'event' command.
+     * @return An AddEventCommand object.
+     * @throws SageException If the arguments are invalid or missing, or date format is incorrect.
+     */
+    private static Command parseEventCommand(String arguments) throws SageException {
+        if (arguments.trim().isEmpty()) {
+            throw new SageException("The description of an event cannot be empty.");
+        }
+        if (!arguments.contains(" /from ") || !arguments.contains(" /to ")) {
+            throw new SageException("Invalid event format. Please use: event <description> /from <d/M/yyyy HHmm> /to <d/M/yyyy HHmm>");
+        }
+        String[] eventParts = arguments.split(" /from ");
+        String[] fromToParts = eventParts[1].split(" /to ");
+        LocalDateTime from = parseDateTime(fromToParts[0]);
+        LocalDateTime to = parseDateTime(fromToParts[1]);
```

Comment:
The variable names here could be a little ambiguous, perhaps we could adjust it?

```suggestion
        String[] timingParts = eventParts[1].split(" /to ");
        LocalDateTime startTime = parseDateTime(fromToParts[0]);
        LocalDateTime endTime = parseDateTime(fromToParts[1]);
```